nuxeo-ai-core/src/main/java/org/nuxeo/ai/enrichment/EnrichmentUtils.java

```diff
@@ -230,13 +239,19 @@ public static Blob convertImageBlob(String service, Blob blob, int width, int he
     /**
      * Put an entry in the enrichment cache, specify the TTL in seconds.
      */
-    public static void cachePut(String cacheKey, Collection<EnrichmentMetadata> metadata, long ttl) {
+    public static void cachePut(String cacheKey, Collection<AIMetadata> metadata, long ttl) {
         if (isNotBlank(cacheKey)) {
             KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(ENRICHMENT_CACHE_KV);
             try {
-                byte[] result = JacksonUtil.MAPPER.writeValueAsBytes(metadata);
-                if (result != null) {
-                    kvStore.put(cacheKey, result, ttl);
+                if (!metadata.isEmpty()) {
+                    if (metadata.stream().allMatch(aiMetadata -> aiMetadata instanceof EnrichmentMetadata)) {
```

Comment:
if I remember correctly,`instance of` can be used in more stream like way `EnrichmentMetadata .class#isInstance`